mirror_base.cpp

```diff
@@ -160,18 +162,54 @@ namespace mirror
 		}
 	}
 
-	void* ClassMember::getInstanceMemberPointer(void* _classInstancePointer) const
+	const mirror::MetaData* MetaDataSet::findMetaData(const char* _key) const
 	{
-		return reinterpret_cast<uint8_t*>(_classInstancePointer) + m_offset;
+		if (_key == nullptr)
+			return nullptr;
+
+		uint32_t hash = HashCString(_key);
+		auto it = m_metaData.find(hash);
+		if (it == m_metaData.end())
+			return nullptr;
+
+		return &it->second;
+	}
+
+	void TypeDesc::setFactory(FactoryFunctionPtr _factory)
+	{
+		m_factory = _factory;
 	}
 
-	mirror::MetaData* ClassMember::getMetaData(const char* _key) const
+	bool TypeDesc::hasFactory() const
 	{
+		return m_factory != nullptr;
+	}
+
+	void* TypeDesc::instantiate() const
+	{
+		if (m_factory != nullptr)
+		{
+			return m_factory();
+		}
 		return nullptr;
```

Comment:
je mettrai presque une assert la. genre: did you check if `hasFactory()` ?

Reply:
Je suis plus trop sûr de ce genre d'asserts. Le travail me fait douter, je me ramollis...
Non mais ça peut peut-être être valide pour le user de checker son pointeur à la sortie et de faire un truc, j'avoue je sais pas trop encore